Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingBasicShaderExecutionReordering/Raytracing.hlsl

```diff
@@ -209,14 +213,12 @@ void MyClosestHitShader(inout RayPayload payload, in MyAttributes attr)
 
         // Sample the texture
         float3 colorSum = float3(0, 0, 0);
-        [unroll]
-        for (int i = 0; i < 18; ++i)
+        for (uint i = 0; i < 1000; ++i)
         {
-            float2 offset = float2(i * 0.01, i * 0.01);
-            colorSum += MaterialTexture.SampleLevel(TextureSampler, uv + offset, 0).rgb;
+            colorSum += MaterialTexture.SampleLevel(TextureSampler, uv, 0).rgb;
+            colorSum = sin(colorSum) + cos(colorSum);
```

Comment:
On my 4090, this colorSum = sin(colorSum) + cos(colorSum); seems to hit an nvidia driver bug, crashing in driver shader compiler.

Splitting to per component solved it:

            colorSum.r = sin(colorSum.r) + cos(colorSum.r);
            colorSum.g = sin(colorSum.g) + cos(colorSum.g);
            colorSum.b = sin(colorSum.b) + cos(colorSum.b);

Though that is a bit ugly.  Could try to get NVIDIA to fix the bug, but it's probably not worth the trouble.

Reply:
Also, it's possible the compiler/driver will be smart and notice it could just sample the texture once, reusing the result.
If the intent is to sample the texture for each loop, could keep the uv offset you had before?